### PR TITLE
update pivnet-cli + go-pivnet

### PR DESCRIPTION
**Pull request by rkoster**
This PR updates go-pivnet which resulted in a pivnet-cli update.
The motivation for this PR is to have https://github.com/pivotal-cf/go-pivnet/pull/26 available.

### CODE REVIEW

**Comment by jtarchie** [comment]
Thanks for taking the time to make the PR.
Can you describe a bit more what this gains us? It is not clear in the code changes.

**Comment by rkoster** [comment]
@jtarchie its just an update of the pivnet related dependencies, which have changed their api slightly. I had to go through these changes my self for https://github.com/starkandwayne/om-tiler which depends on `om` as well as `go-pivnet`. Without these changes I would not be able to update `go-pivnet` in `om-tiler`.

File: configtemplate/metadata/pivnet.go [hunk 1]
```diff
@@ -177,28 +181,24 @@ func productFileKeysByGlobs(
...
-		var err error
-		accessToken, err := tokenFetcher.GetToken()
-
-		if err != nil {
-			log.Fatalf("Exiting with error: %s", err)
-		}
-		req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", accessToken))
-	} else {
-		req.Header.Add("Authorization", fmt.Sprintf("Token %s", a.ClientConfig.Token))
+	token, err := a.TokenService.AccessToken()
```
Comment (rkoster) [inline configtemplate/metadata/pivnet.go]: main difference is, we now need to create a TokenService which is in charge of dealing with the AccessToken.

File: configtemplate/metadata/pivnet.go [hunk 2]
```diff
@@ -177,28 +181,24 @@ func productFileKeysByGlobs(
 
 type AuthenticatedPivnetClient struct {
 	ClientConfig pivnetapi.ClientConfig
+	TokenService pivnetapi.AccessTokenService
 	HTTPClient   *http.Client
 }
 
 func (a AuthenticatedPivnetClient) Do(req *http.Request) (*http.Response, error) {
-	const legacyAPITokenLength = 20
-	if len(a.ClientConfig.Token) > legacyAPITokenLength {
```
Comment (rkoster) [inline configtemplate/metadata/pivnet.go]: All this logic is now taken care of by the TokenService and the `AuthorizationHeader` function.

File: download_clients/pivnet_client.go [hunk 3]
```diff
@@ -180,7 +182,10 @@ func (p *pivnetClient) getLatestStemcell(dependencies []pivnet.ReleaseDependency
 	return stemcellSlug, stemcellVersion, nil
 }
 
-const errorForVersion = "versioning of stemcell dependency in unexpected format: \"major.minor\" or \"major\". the following version could not be parsed: %s"
+const (
+	errorForVersion = "versioning of stemcell dependency in unexpected format: \"major.minor\" or \"major\". the following version could not be parsed: %s"
+	userAgent       = "om-download-product"
```
Comment (rkoster) [inline download_clients/pivnet_client.go]: created a constant because AccessTokenService and pivnet.ClientConfig both need a userAgent

**Comment by rkoster** [comment]
have left some comments [here](https://github.com/pivotal-cf/om/pull/395/files) with some additional context on the changes.

**Comment by jtarchie** [comment]
I've validate that this PR works.

Tested by:
```
$ go run main.go config-template --output-directory /tmp --pivnet-api-token "new-token" --pivnet-product-slug elastic-runtime --product-version 2.5.1 --product-file-glob '*srt*'

$ go run main.go config-template --output-directory /tmp --pivnet-api-token "old-style-token" --pivnet-product-slug elastic-runtime --product-version 2.5.0 --product-file-glob '*srt*'

$ ls /tmp/cf
2.5.0 2.5.1
```

**Comment by jtarchie** [comment]
Merged and rebased, so SHAs aren't that same.

**Comment by rkoster** [comment]
Thanks for merging 👍

**Pull request closed without merging**